language-plutus-core/src/Language/PlutusCore/Normalize.hs

```diff
@@ -29,16 +29,16 @@ runNormalizeTypeM a = liftQuote $ runReaderT a (TypeEnv mempty)
 
 -- | Locally extend a 'TypeEnv' in a 'NormalizeTypeM' computation.
 withExtendedTypeEnv
-    :: HasUnique (tyname ())
+    :: HasUnique (tyname ()) TypeUnique
     => tyname () -> NormalizedType tyname () -> NormalizeTypeM tyname a -> NormalizeTypeM tyname a
 withExtendedTypeEnv name ty =
-    local (TypeEnv . IntMap.insert (unUnique $ name ^. unique) ty . unTypeEnv)
+    local (TypeEnv . IntMap.insert (name ^. unique . coerced) ty . unTypeEnv)
```

Comment:
Perhaps we should have a `UniqueMap` that's a wrapper around `IntMap` and handles some of this stuff, but that can happen later.

Reply:
Actually, `UniquesRenaming` is such a thing. We just need to generalize the name and put this stuff into `Language.PlutusCore.Unique` together with the definitions of various uniques. I'll do this in a separate PR.